project/src/main/puzzle/level/level.gd

```diff
@@ -11,15 +11,22 @@ This class needs to be a node because it provides utility methods which utilize
 # emitted after the level has customized the puzzle's settings.
 signal settings_changed
 
+# emitted when the 'level_state' field changes, such as when starting or clearing a level.
+signal level_state_changed
+
+enum LevelState {
+	NONE,
+	BEFORE, # the level hasn't been started
```

Comment:
Maybe clarify this a bit; `NONE` = the level hasn't been launched yet; `BEFORE` = level has been launched but hasn't been played yet

Reply:
`AFTER` = the level has been finished successfully

Reply:
Updated enum constants to be more descriptive; `the level hasn't been launched`